src/core/resource.cc

```diff
@@ -212,14 +212,15 @@ namespace resource {
 size_t calculate_memory(const configuration& c, size_t available_memory, float panic_factor = 1) {
     size_t default_reserve_memory = std::max<size_t>(1536 * 1024 * 1024, 0.07 * available_memory) * panic_factor;
     auto reserve = c.reserve_memory.value_or(default_reserve_memory);
+    reserve += c.reserve_additional_memory;
     size_t min_memory = 500'000'000;
     if (available_memory >= reserve + min_memory) {
         available_memory -= reserve;
     } else {
         // Allow starting up even in low memory configurations (e.g. 2GB boot2docker VM)
         available_memory = min_memory;
     }
-    size_t mem = c.total_memory.value_or(available_memory);
+    size_t mem = c.total_memory ? *c.total_memory - c.reserve_additional_memory : available_memory;
```

Comment:
Why only substract reserve_additional_memory from explicitly configured memory? If seastar is allowed to use all it wants (i.e. -- no -m is given), why does it make any difference in terms of

> Now we subtract the extra memory from the total app memory so the actual memory used by scylla is still limited by this option

?

Reply:
We subtract the extra memory in both cases, whether `-m` was given or not (we want this `mem` variable here to be lower) - the subtraction happens either here in line 223 (if `-m` is given) or as a result of lines 215 and 218 (if `-m` is not given). 
I omitted the low memory configuration case (the 500MB) because in that case we aren't obeying our specified `--reserve-memory` anyway, so we aren't breaking any assumptions by leaving even less memory for the system and if we assume that an app may need 500MB for the seastar allocations, it will probably keep needing it. But if it makes more sense to reserve the memory in this case as well, we can do it.

Reply:
> We subtract the extra memory in both cases, whether -m was given or not (we want this mem variable here to be lower) - the subtraction happens either here in line 223 (if -m is given) or as a result of lines 215 and 218 (if -m is not given).

Ah, indeed :exploding_head: